src/codegen/compile.ml

```diff
@@ -2706,17 +2708,18 @@ module Object = struct
 
 
   (* Returns a pointer to the object field (without following the indirection) *)
-  let idx_hash_raw env =
-    Func.share_code2 env "obj_idx" (("x", I32Type), ("hash", I32Type)) [I32Type] (fun env get_x get_hash ->
+  let idx_hash_raw env m = (* HERE *)
+    let name = Printf.sprintf "obj_idx<%d>" m  in
```

Comment:
Hmm, quite wasteful duplicating that code for every possible starting index, when the body of the function really just differs in one consant.

I suggest to write
```
Func.share_code3 env "obj_idx
```
and pass the starting index as the second parameter instead.

Reply:
I totally get what you are saying, but I am afraid I don't agree... I posted this concern in the description. For small Wasm we'd have wasteful duplication, but who cares, we don't cross the message limit. For big programs it is likely that a lot of field indexing is happening, so the 2 bytes of Wasm _for each call site_ will start to become relevant, potentially outweighing the penalty in Wasm coming  from the duplication. Of course we could have short functions for each start index which then (tail-)call the `obj_idx` that you propose. This adds some double-call overhead.

Reply:
Hmm, I see.

Still only gut feeling, but a common function called from everywhere is probably also better for the runtime’s inliner: It’ll be hot code faster.

Also, why stop here? Why not include the `hash` (which is also statically known) in the function name and have static functions for all combinations.

Reply:
Next PR :-) Good points about many small stubs.

Reply:
That's weird, I though I had made the same suggestion as @nomeata about abstracting on the start index to avoid duplication but the comment seems to have disappeared. GH acting strangely or did I forget to commit?

Reply:
GH's fault :->

Reply:
I am working on a follow-up PR to minimise duplication, based on the ideas and feedback. I prefer to merge this as-is.